<launch>

  <node pkg="line_detection" name="backproject_lanedetection" type="backproject.py" output="screen">
    
    <param name="use_mono" type="bool" value="false" />
    <param name="use_compressed_format" type="bool" value="true" />
    <param name="subscriber_image_topic" type="str" value="/camera/image_raw" />
    <param name="publisher_image_topic" type="str" value="/line_image" />
    <param name="training_file_name" type="str" value="training_for_backprojection_1.png" />
    <param name="buffer_size" type="int" value="52428800" />

  </node>

  <node pkg="line_detection" name="brightestpixel_lanedetection" type="brightest-pixel.py" output="screen">
    
    <param name="use_mono" type="bool" value="true" />
    <param name="use_compressed_format" type="bool" value="true" />
    <param name="subscriber_image_topic" type="str" value="/backproject_lanedetection/line_image" />
    <param name="publisher_image_topic" type="str" value="/line_image" />
    <param name="training_file_name" type="str" value="training_for_backprojection_1.png" />
    <param name="buffer_size" type="int" value="52428800" />

  </node>

  <node pkg="line_detection" name="dilate_lanedetection" type="dilate.py" output="screen">
    
    <param name="use_mono" type="bool" value="true" />
    <param name="use_compressed_format" type="bool" value="true" />
    <param name="subscriber_image_topic" type="str" value="/brightestpixel_lanedetection/line_image" />
    <param name="publisher_image_topic" type="str" value="/line_image" />
    <param name="training_file_name" type="str" value="training_for_backprojection_1.png" />
    <param name="buffer_size" type="int" value="52428800" />

  </node>

  <node pkg="line_detection" name="lswms_lanedetection" type="lswms" output="screen">
    <param name="~image_transport" value="compressed" />
    <remap from="/lswms_lanedetection/subscriber/compressed" to="dilate_lanedetection/line_image/compressed"/>
  </node>

  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen" />
  
</launch>